plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -234,48 +239,64 @@ def print_events(events)
       end
     end
 
-    def unstable!(reason, bad_release_statuses)
+    def unstable!(reason, bad_resource_statuses)
       @output.puts "UNSTABLE: #{reason}"
-      bad_release_statuses.select(&:pod).each do |status|
-        @output.puts "  #{pod_identifier(status.pod)}: #{status.details}"
+      bad_resource_statuses.select(&:resource).each do |status|
+        @output.puts "  #{resource_identifier(status)}: #{status.details}"
       end
     end
 
-    def release_statuses(pods, release_doc)
-      group = release_doc.deploy_group
+    def resource_statuses(pods, release_doc)
+      deploy_group = release_doc.deploy_group
       role = release_doc.kubernetes_role
 
-      pods = pods.select { |pod| pod.role_id == role.id && pod.deploy_group_id == group.id }
-
-      statuses = Array.new(release_doc.desired_pod_count).each_with_index.map do |_, i|
-        pod = pods[i]
+      # monitor each resource we deploy directly
+      resource_statuses = release_doc.resources.map do |resource|
```

Comment:
can we split up this method, a separate one for monitoring directly deployed resources, and monitoring pods we expect to appear?